I want to clean the sask df I have to take the sensor dict column and clean it so its easy to analyze

In [8]:
import pandas as pd
import geopandas as gpd
import ast # used to convert string representation of list to actual list

In [6]:
# import api1/saskatoon_sensors.csv
df = pd.read_csv("/Users/dawsonconlon/Coding/Python/GIS_API/api1/saskatoon_sensors.csv")
print(df.head())

    id       name      locality country.name  coordinates.latitude  \
0  958  Saskatoon  SASKATCHEWAN       Canada             52.136055   

   coordinates.longitude                                            sensors  
0            -106.662796  [{'id': 5077666, 'name': 'no2 ppm', 'parameter...  


In [9]:
df["sensors"] = df["sensors"].apply(ast.literal_eval)

In [2]:
rows = []

for _, row in df.iterrows():
    for sensor in row["sensors"]:
        rows.append({
            "station_id": row["id"],
            "station_name": row["name"],
            "locality": row["locality"],
            "latitude": row["coordinates.latitude"],
            "longitude": row["coordinates.longitude"],
            "pollutant": sensor["parameter"]["displayName"],
            "parameter": sensor["parameter"]["name"],
            "units": sensor["parameter"]["units"]
        })

clean_df = pd.DataFrame(rows)

NameError: name 'df' is not defined

In [12]:
print(clean_df.head())

   station_id station_name      locality   latitude   longitude pollutant  \
0         958    Saskatoon  SASKATCHEWAN  52.136055 -106.662796       NO₂   
1         958    Saskatoon  SASKATCHEWAN  52.136055 -106.662796        O₃   
2         958    Saskatoon  SASKATCHEWAN  52.136055 -106.662796     PM2.5   

  parameter  units  
0       no2    ppm  
1        o3    ppm  
2      pm25  µg/m³  


It has occured to me that the api I pulled simly has sensor location and we found the sensor ID that we want (958) but we need to pull a time series measurment from this sensor with actual measurments, so we have to just slightly adjust the api and then try aging